pinot-common/src/main/java/org/apache/pinot/common/function/scalar/DataTypeConversionFunctions.java

```diff
@@ -129,4 +129,27 @@ public static byte[] base64Encode(byte[] input) {
   public static byte[] base64Decode(byte[] input) {
     return Base64.getDecoder().decode(input);
   }
+
+  /**
+   * Converts a hex string to the corresponded long value.
+   * @param input hex decimal string
+   * @return decoded long value
+   */
+  @ScalarFunction
+  public static long hexDecimalToLong(String input) {
+    if (input.startsWith("0x")) {
+      input = input.substring(2);
+    }
+    return Long.parseLong(input, 16);
```

Comment:
Tip: You can call `public static long parseLong(CharSequence s, int beginIndex, int endIndex, int radix)` instead allocating a new string here.

Reply:
👍  updated the PR